### PR TITLE
[otelcol] Remove all default providers/converters

### PR DESCRIPTION
**Pull request by TylerHelmuth**
#### Description
Removed deprecated `NewCommand`.  Deprecates `NewCommandMustSetProvider`.  Adds `NewCommand` that requires at least one provider be set.

Removes usage of imported providers/converters from otelcol (but they still live in otelcoltest until https://github.com/open-telemetry/opentelemetry-collector/pull/10417 is merged.

#### Link to tracking issue
closes https://github.com/open-telemetry/opentelemetry-collector/issues/10290.

#### Testing
Unit tests

### CODE REVIEW

**Review by mx-psi — COMMENTED** [review]
Can we file an issue on the Jaeger project and ping Yuri there so that they are aware of this change?

**Comment by TylerHelmuth** [comment]
I'm planning to submit another PR to jaeger to handle the deprecation process, was waiting for the 0.103.0 release to finish in Contrib.

**Review by mx-psi — APPROVED** [review]
Looks like there are some tests to fix, but otherwise this LGTM. I would like to merge this after Jaeger is fixed

**Comment by TylerHelmuth** [comment]
Jaeger is updated: https://github.com/jaegertracing/jaeger/pull/5657

File: cmd/builder/internal/builder/main_test.go
Comment (TylerHelmuth) [inline cmd/builder/internal/builder/main_test.go:261]: These tests all pass in a list of providers, and therefore depend on those providers in the go mod list.  The builder templates only add them when `.Distribution.SupportsConfmapFactories` is true.  That value was not being set so all these tests should have been failing.  They were saved instead by the fact that otelcol depended on the providers.  The changes in this PR remove those dependencies when `otelcoltest` is not being used, so these tests started failing.  The fix was to make sure `.Distribution.SupportsConfmapFactories` is true, which is included in `cfg.SetBackwardsCompatibility()`.

Comment (evan-bradley) [reply cmd/builder/internal/builder/main_test.go:261]: Nice catch, thanks for the explanation.

File: otelcol/command.go
Comment (evan-bradley) [inline otelcol/command.go:52]: We should probably have a deprecation changelog entry for this, as inconsequential as it likely is.

File: otelcol/command.go
Comment (evan-bradley) [inline otelcol/command.go:73]:
```suggestion
	if len(resolverSet.ProviderFactories) == 0 {
```
I still think we should remove this check and have it handled by the resolver, but if we intend to keep it here, we should probably remove the converter check now that we don't need to look for a "zero-valued" factory config.

**Comment by mx-psi** [comment]
@TylerHelmuth can you resolve the conflict?

**Comment by TylerHelmuth** [comment]
@mx-psi conflict handled.

**Comment by mx-psi** [comment]
@evan-bradley will merge this by Friday EU morning unless you comment by then!

**Comment by evan-bradley** [comment]
Sorry, I was out last week, thanks for not holding this up on me. This looks good to me, the negative diff in the go.mod file is a win in my eyes.

**Comment by mx-psi** [comment]
Agreed, I am pretty happy about the `go.mod` update :)

**Pull request merged**
